core/tee/tee_svc.c

```diff
@@ -815,9 +831,15 @@ static TEE_Result tee_svc_update_out_param(
 {
 	size_t n;
 	uint64_t *vals = usr_param->vals;
-	size_t sz = 0;
+	uint64_t sz = 0;
+	uint64_t in_sz = 0;
+	uint64_t val_buf[2] = { };
 
 	for (n = 0; n < TEE_NUM_PARAMS; n++) {
+		TEE_Result res = TEE_SUCCESS;
+		val_buf[0] = 0;
```

Comment:
It would be better to move `uint64_t val_buf[2] = { };` above here.

Reply:
I like it. Updated